### PR TITLE
Code coverage - DO NOT MERGE

### PR DESCRIPTION
**Pull request by bansisoni**

### CODE REVIEW

**Comment by cortx-admin** [comment]
<h1>Jenkins CI Result : <a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184/>Motr#184</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>12</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/01motr-single-node/00userspace-tests target=_blank>01motr-single-node/00userspace-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/01motr-single-node/03initscript-tests target=_blank>01motr-single-node/03initscript-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/01motr-single-node/28motr-sys-kvs-test target=_blank>01motr-single-node/28motr-sys-kvs-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/02motr-single-node/20pool-version-test target=_blank>02motr-single-node/20pool-version-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/02motr-single-node/02system-tests target=_blank>02motr-single-node/02system-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/02motr-single-node/51kem target=_blank>02motr-single-node/51kem</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/03motr-single-node/54sns-repair-motr-1f target=_blank>03motr-single-node/54sns-repair-motr-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/03motr-single-node/17sss-st-test target=_blank>03motr-single-node/17sss-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/03motr-single-node/58sns-repair-motr-ios-fail target=_blank>03motr-single-node/58sns-repair-motr-ios-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/03motr-single-node/60sns-repair-motr-abort-quiesce target=_blank>03motr-single-node/60sns-repair-motr-abort-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/04motr-single-node/25spiel-sns-repair-test target=_blank>04motr-single-node/25spiel-sns-repair-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/04motr-single-node/26spiel-sns-repair-quiesce-test target=_blank>04motr-single-node/26spiel-sns-repair-quiesce-test</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>1</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/01motr-single-node/37protocol target=_blank>01motr-single-node/37protocol</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>26</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/01motr-single-node/46m0crate target=_blank>01motr-single-node/46m0crate</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/01motr-single-node/01kernel-tests target=_blank>01motr-single-node/01kernel-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/01motr-single-node/30hagen-test target=_blank>01motr-single-node/30hagen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/01motr-single-node/29confgen-test target=_blank>01motr-single-node/29confgen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/01motr-single-node/35m0singlenode target=_blank>01motr-single-node/35m0singlenode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/01motr-single-node/04net-tests target=_blank>01motr-single-node/04net-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/01motr-single-node/05rpcping-tests target=_blank>01motr-single-node/05rpcping-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/01motr-single-node/06console-tests target=_blank>01motr-single-node/06console-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/01motr-single-node/52singlenode-sanity target=_blank>01motr-single-node/52singlenode-sanity</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/02motr-single-node/21spiel-test target=_blank>02motr-single-node/21spiel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/02motr-single-node/47motr-tests-user-kernel target=_blank>02motr-single-node/47motr-tests-user-kernel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/02motr-single-node/09poolmach-tests target=_blank>02motr-single-node/09poolmach-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/02motr-single-node/41motr-conf-update target=_blank>02motr-single-node/41motr-conf-update</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/02motr-single-node/08degraded-mode-tests target=_blank>02motr-single-node/08degraded-mode-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/02motr-single-node/24m0d-fsync-test target=_blank>02motr-single-node/24m0d-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/02motr-single-node/22rpc-cancel-test target=_blank>02motr-single-node/22rpc-cancel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/02motr-single-node/53clusterusage-alert target=_blank>02motr-single-node/53clusterusage-alert</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/02motr-single-node/23m0d-signal-test target=_blank>02motr-single-node/23m0d-signal-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/03motr-single-node/59sns-repair-motr-abort target=_blank>03motr-single-node/59sns-repair-motr-abort</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/03motr-single-node/55sns-repair-motr-1n-1f target=_blank>03motr-single-node/55sns-repair-motr-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/03motr-single-node/56sns-repair-motr-mf target=_blank>03motr-single-node/56sns-repair-motr-mf</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/03motr-single-node/61spiel-multi-confd target=_blank>03motr-single-node/61spiel-multi-confd</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/03motr-single-node/16conf-st-test target=_blank>03motr-single-node/16conf-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/03motr-single-node/57sns-repair-motr-1k-1f target=_blank>03motr-single-node/57sns-repair-motr-1k-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/03motr-single-node/36spare-reservation target=_blank>03motr-single-node/36spare-reservation</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit/04motr-single-node/19sns-abort-test target=_blank>04motr-single-node/19sns-abort-test</a></p></details></td></tr><tr><td>Total</td><td>39</td><td><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/184//testReport/junit>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

File: pool/pool.c [hunk 1]
```diff
@@ -1671,7 +1671,7 @@ M0_INTERNAL int m0_pool_version_append(struct m0_pools_common  *pc,
 				       struct m0_conf_pver     *pver,
 				       struct m0_pool_version **pv)
 {
-	struct m0_conf_pool *cp;
+	struct m0_conf_pool *cp = NULL;
```
Comment (huanghua78) [inline pool/pool.c:1674]:
```
M0_ASSERT(cp != NULL); 
```
should be added before pool_find(), otherwise it might panic there.

**Comment by cortx-admin** [comment]
<h1>Jenkins CI Result : <a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204/>Motr#204</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>4</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/01motr-single-node/03initscript-tests target=_blank>01motr-single-node/03initscript-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/02motr-single-node/51kem target=_blank>02motr-single-node/51kem</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/03motr-single-node/17sss-st-test target=_blank>03motr-single-node/17sss-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/04motr-single-node/27sns-repair-io-fail-test target=_blank>04motr-single-node/27sns-repair-io-fail-test</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>1</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/01motr-single-node/37protocol target=_blank>01motr-single-node/37protocol</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>35</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/01motr-single-node/46m0crate target=_blank>01motr-single-node/46m0crate</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/01motr-single-node/01kernel-tests target=_blank>01motr-single-node/01kernel-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/01motr-single-node/00userspace-tests target=_blank>01motr-single-node/00userspace-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/01motr-single-node/30hagen-test target=_blank>01motr-single-node/30hagen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/01motr-single-node/29confgen-test target=_blank>01motr-single-node/29confgen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/01motr-single-node/35m0singlenode target=_blank>01motr-single-node/35m0singlenode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/01motr-single-node/04net-tests target=_blank>01motr-single-node/04net-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/01motr-single-node/05rpcping-tests target=_blank>01motr-single-node/05rpcping-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/01motr-single-node/28motr-sys-kvs-test target=_blank>01motr-single-node/28motr-sys-kvs-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/01motr-single-node/06console-tests target=_blank>01motr-single-node/06console-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/01motr-single-node/52singlenode-sanity target=_blank>01motr-single-node/52singlenode-sanity</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/02motr-single-node/20pool-version-test target=_blank>02motr-single-node/20pool-version-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/02motr-single-node/21spiel-test target=_blank>02motr-single-node/21spiel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/02motr-single-node/47motr-tests-user-kernel target=_blank>02motr-single-node/47motr-tests-user-kernel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/02motr-single-node/09poolmach-tests target=_blank>02motr-single-node/09poolmach-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/02motr-single-node/41motr-conf-update target=_blank>02motr-single-node/41motr-conf-update</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/02motr-single-node/08degraded-mode-tests target=_blank>02motr-single-node/08degraded-mode-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/02motr-single-node/24m0d-fsync-test target=_blank>02motr-single-node/24m0d-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/02motr-single-node/02system-tests target=_blank>02motr-single-node/02system-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/02motr-single-node/22rpc-cancel-test target=_blank>02motr-single-node/22rpc-cancel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/02motr-single-node/53clusterusage-alert target=_blank>02motr-single-node/53clusterusage-alert</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/02motr-single-node/23m0d-signal-test target=_blank>02motr-single-node/23m0d-signal-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/03motr-single-node/54sns-repair-motr-1f target=_blank>03motr-single-node/54sns-repair-motr-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/03motr-single-node/59sns-repair-motr-abort target=_blank>03motr-single-node/59sns-repair-motr-abort</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/03motr-single-node/58sns-repair-motr-ios-fail target=_blank>03motr-single-node/58sns-repair-motr-ios-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/03motr-single-node/60sns-repair-motr-abort-quiesce target=_blank>03motr-single-node/60sns-repair-motr-abort-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/03motr-single-node/55sns-repair-motr-1n-1f target=_blank>03motr-single-node/55sns-repair-motr-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/03motr-single-node/56sns-repair-motr-mf target=_blank>03motr-single-node/56sns-repair-motr-mf</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/03motr-single-node/61spiel-multi-confd target=_blank>03motr-single-node/61spiel-multi-confd</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/03motr-single-node/16conf-st-test target=_blank>03motr-single-node/16conf-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/03motr-single-node/57sns-repair-motr-1k-1f target=_blank>03motr-single-node/57sns-repair-motr-1k-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/03motr-single-node/36spare-reservation target=_blank>03motr-single-node/36spare-reservation</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/04motr-single-node/25spiel-sns-repair-test target=_blank>04motr-single-node/25spiel-sns-repair-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/04motr-single-node/19sns-abort-test target=_blank>04motr-single-node/19sns-abort-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit/04motr-single-node/26spiel-sns-repair-quiesce-test target=_blank>04motr-single-node/26spiel-sns-repair-quiesce-test</a></p></details></td></tr><tr><td>Total</td><td>40</td><td><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/204//testReport/junit>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

**Comment by huanghua78** [comment]
Hello @bansisoni ,
Please check if this PR is still valid and needed.
If not, please close it.
If yes, please update the status.

**Comment by bansisoni** [comment]
> Hello @bansisoni ,
> Please check if this PR is still valid and needed.
> If not, please close it.
> If yes, please update the status.

This PR is not valid anymore. Code changes for compile time errors are fixed and merged in another PR. Closing this PR.
Motr coverage CI is using this code_coverage branch to get updated motr-single_tests.yaml and build script. So, branch still needed.

**Pull request closed without merging**